src/components/Facility/DischargedPatientsList.tsx

```diff
@@ -136,6 +213,32 @@ const DischargedPatientsList = ({
     }
   }, [qParams]);
 
+  const dateRangeFields = [
+    [params.created_date_before, params.created_date_after],
+    [params.modified_date_before, params.modified_date_after],
+    [params.date_declared_positive_before, params.date_declared_positive_after],
+    [params.last_vaccinated_date_before, params.last_vaccinated_date_after],
+    [
+      params.last_consultation_encounter_date_before,
+      params.last_consultation_encounter_date_after,
+    ],
+    [
+      params.last_consultation_discharge_date_before,
+      params.last_consultation_discharge_date_after,
+    ],
+  ];
+
+  const isExportAllowed = dateRangeFields.some(([startDate, endDate]) => {
+    const days = calculateDaysBetween(startDate, endDate);
+    return days > 0 && days <= 7;
+  });
+
+  const { refetch: exportDischargedAsCSV } = usePatientExport(
+    routes.listFacilityDischargedPatients,
+    { ...params },
+    { facility_external_id },
+  );
+
   const { data: districtData } = useTanStackQueryInstead(routes.getDistrict, {
```

Comment:
Few more of these - switch to useQuery.

Other than that, looks good.

Currently doesn't support filtering by discharge date range (BE blocker; will need to open an issue there).

Reply:
@Jacobjeevan  Replaced all queries using useQuery Hook .And Should I remove the filtering by discharge date or let it be

Reply:
@Jacobjeevan Could you check the changes made and update the label